File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums, k) {
        const seen = {0: 1};
        
        let prefixSum = 0;
        let res = 0;
        for(const num of nums) {
            prefixSum += num;
            const complement = prefixSum - k;
            res += seen[complement] ?? 0;
            seen[prefixSum] = 1 + (seen[prefixSum] ?? 0);
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)

        if not nums:
            return 0

        longest = 1
        for num in nums:
            count = 1
            next_ele = num + 1
        
            while(next_ele in seen):
                count += 1
                next_ele += 1
            
            longest = max(count, longest)
        
        return longest